packages/loopring_v3.js/src/request_processors/spot_trade_processor.ts

```diff
@@ -116,6 +125,15 @@ export class SpotTradeProcessor {
         .balance.iadd(s.fillBB)
         .isub(s.feeB);
 
+      // virtual balances
+      if (
+        accountB.getBalance(tokenA).weightAMM.gt(new BN(0)) &&
```

Comment:
I didn't update all references of "weight" to "virtual balance" because I think that would bring too many unnecessary changes to both the protocol and the relayer (would change the block format passed into the prover etc...), and so seemed more error prone than just to keep the name for all existing code. "weight" also not really that bad of a name even in the new system because it's still the weight of the token in the pool, just used a bit differently.